File: app/src/main/java/com/example/moviesearch/model/Movie.java
package com.example.moviesearch.model;

import com.google.gson.annotations.SerializedName;

public class Movie {
    @SerializedName(value = "Title")
    private String title;
    @SerializedName(value = "Year")
    private String year;
    @SerializedName(value = "Poster")
    private String poster;
    private String imdbID;
    /* 封装的原则：通过 getter 和 setter 方法来访问和修改这些
    私有变量，从而实现数据的保护，防止外部代码直接修改它们 */

    public Movie(String title, String year, String poster, String imdbID) {
        this.title = title;
        this.year = year;
        this.poster = poster;
        this.imdbID = imdbID;
    }
    //this用于引用当前对象的实例，区分参数和成员变量

    //Getter方法，用于获取属性值
    public String getTitle() {
        return title;
    }

    public String getYear() {
        return year;
    }

    public String getPoster() {
        return poster;
    }

    public String getImdbID() {
        return imdbID;
    }
}

File: Movie/app/src/main/java/com/example/movie/model/Movie.kt
package com.example.movie.model

import com.google.gson.annotations.SerializedName

data class Movie (
    val imdbID: String,
    @SerializedName("Title") val title: String,
    @SerializedName("Poster") val poster: String,
    @SerializedName("Year") val year: String?,
    @SerializedName("Genre") val genre: String?,
    @SerializedName("Director") val director: String?,
    @SerializedName("Country") val country: String?,
)

// 在 Kotlin 中，构造函数参数会自动成为类的属性，且不需要显式地定义 getter 和 setter 方法